src/factory/fetcher-factory.js

```diff
@@ -0,0 +1,131 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
```

Comment:
Ah, I see. In the fullness of time there will be multiple fetchers (e.g, npm, nuget, ...) and each will have its own config info.  So in fact the options name is "wrong" and should be more specific (e.g., githubFetcher).